source/connection-monitoring-and-pooling/tests/pool-clear-close-in-use.json

```diff
@@ -0,0 +1,69 @@
+{
+  "version": 1,
+  "style": "unit",
+  "description": "Pool clear MUST support lazily closing in use connections",
+  "poolOptions": {
+    "backgroundThreadIntervalMS": 100
+  },
+  "operations": [
+    {
+      "name": "ready"
+    },
+    {
+      "name": "checkOut"
+    },
+    {
+      "name": "checkOut"
+    },
+    {
+      "name": "clear",
+      "closeInUseConnections": true
+    },
+    {
+      "name": "waitForEvent",
+      "event": "ConnectionClosed",
+      "count": 2,
+      "timeout": 1000
+    },
+    {
+      "name": "close"
+    }
+  ],
+  "events": [
+    {
+      "type": "ConnectionCheckedOut",
+      "connectionId": 1,
+      "address": 42
+    },
+    {
+      "type": "ConnectionCheckedOut",
+      "connectionId": 2,
+      "address": 42
+    },
+    {
+      "type": "ConnectionPoolCleared",
```

Comment:
As @DmitryLukyanov noticed, ordering of `ConnectionPoolCleared` and `ConnectionClosed` is not guaranteed here. 
We could validate those events with `waitForEvent` and remove them from expected events, or ignore the `ConnectionPoolCleared` event.
cc @patrickfreed

Reply:
Yeah I think removing the connection pool cleared event from expected events SGTM.

As a side note: should we introduce a new "reason" for in use connections that are closed? Something like "cancelled"?

Reply:
in c#, we don't have such reason yet :) But yeah, maybe it makes sense to distinguish such events, but since in both cases events are "stale", maybe adding a new property like "ClosedInUse" or something?

Reply:
or maybe a reason like `cancelledStale`?

Reply:
I removed asserting on pool clear event in json tests.

Reply:
I think it would be better to reuse the "reason", since this new field would only ever be present when "reason: stale". I think "stale" doesn't quite describe the reason though, since the connection was in use but was actively closed. So far I think "cancelled" is the most accurate, but definitely open to other suggestions.

That does bring up a good point that we should probably add a `closeInUseConnections` property to `PoolClearedEvent`.

Thoughts @BorisDog?

Reply:
At least for c# `cancelled` can have meaning that this connection was cancelled by user via standard .net machinery (CancellationToken) which is a bit ambiguous . Additionally, I think the key point here that we cancel not a random connection, but connection that we consider as unhealthy, ie stale, but I will rely on you both to check a better reason

Reply:
Perhaps its more on the `CommandFailedEvent` to report the cancellation, which would mean `stale` would be fine to keep here. So yeah, using `stale` as the reason SGTM.